nms/app/packages/magmalte/app/components/cwf/CWFGateways.js

```diff
@@ -282,4 +255,87 @@ function CWFGateways(props: WithAlert & {}) {
   );
 }
 
+function GatewayRow(props: {
+  gateway: cwf_gateway,
+  haPairs: cwf_ha_pair[],
+}) {
+  const {gateway, haPairs} = props;
+  const [expanded, setExpanded] = useState<Set<string>>(new Set());
+  const classes = useStyles();
+  const {match, history, relativePath, relativeUrl} = useRouter();
+
+  const gatewayHaPair = haPairs.filter(haPair => {
+    return haPair.gateway_id_1 === gateway.id ||
+        haPair.gateway_id_2 === gateway.id
+  });
+
+  const isGatewayInHaPair = gatewayHaPair.length > 0;
+  const isPrimary = isGatewayInHaPair &&
```

Comment:
minor nit: you can do this gatewayHaPair?.[0] instead of doing isGatewayInHaPair && gatewayHaPair[0]